### ISSUE TITLE
Prefect cloud login should fail expired keys

### ISSUE THREAD

**Issue by EmilRex** [PrefectHQ/prefect#10510]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

In the case where an API key is already set in your profile, running `prefect cloud login -k $API_KEY` with the same `API_KEY` will result in the following message, regardless of whether the key is valid or invalid:

```
This profile is already authenticated with that key.
```

Running the same command with an invalid key without it already being set in your profile results in the following error:
```
Unable to authenticate with Prefect Cloud. Please ensure your credentials are correct.
```

### Describe the proposed behavior

We should always verify that the given API key is valid, even in the case where it's already configured. This can help prevent users from accidentally using an expired or otherwise invalid API key.

**Comment by serinamarie** [comment]
Hi @EmilRex, it's possible that #8641 will close this issue, as it is no longer going to display:
`This profile is already authenticated with that key.` and it should then check first that the key is valid before exiting with success or error. I would be curious if the same issue is faced from that branch? For now I'll add it to the backlog.

**Comment by EmilRex** [comment]
Ah, yep, sounds like that PR absolutely addresses this issue. Specifically the `A PREFECT_API_KEY env var is set and it is the same as a user-provided key but the key is invalid` scenario. Nice! I didn't see that PR, thanks for flagging!

**Comment by serinamarie** [comment]
This is closed as completed by #8641

**Comment by VadymShkarbul** [comment]
Hey! In 2.12 again this problem!
